### ISSUE TITLE
mySQL connection

### ISSUE THREAD

**Issue by dutchguy69** [gotify/server#331]
**Have you read the documentation?**
- [ ] Yes, but it does not include related information regarding my question.
- [ ] Yes, but the steps described in the documentation do not work on my machine.
- [X ] Yes, but I am having difficulty understanding it and wants clarification.

**You are setting up gotify in**
- [ ] Docker
- [X ] Linux native platform
- [ ] Windows native platform

**Describe your problem**

I'm having difficulties setting up my mySQL connection.
I have gotify working with SQLite but want to switch to mySQL db.

I changed the entry in the config file. But I'm not entirely understanding whether I need to change anything there or not. Right now the line is 

  dialect: mysql
  connection: gotify:secret@/gotifydb?charset=utf8&parseTime=True&loc=Local

I read the line as user=gotify, password=secret and database=gotifydb

Before I started the server I created the database and user
CREATE DATABASE gotify;
CREATE USER 'gotify'@'localhost' IDENTIFIED BY 'secret';
GRANT SELECT, INSERT, UPDATE, DELETE ON gotify.* TO 'gotify'@'localhost';

The mySQL instance is local and is on standard port 3306 and I can logon through mySQL client into the database;

It is my understanding that when I start the server the tables are created by the process.

When I start the server I get the following output

Starting Gotify version 2.0.16@2020-05-09-16:00:37
panic: dial tcp 127.0.0.1:3306: connect: connection refused

the localhost is fine and port is fine but clearly the server can't connect and I do not understand it. I do not see connect attempts in the mySQL logs

I've seen other people mentioning the above issue as well but most were running from docker and the connect string seems quite different in that situation.

I have tried to figure out whether it was possible to get some more verbosity from the server but failed in doing so.

Is my connection string wrong.

**Any errors, logs, or other information that might help us identify your problem**

Ex: `docker-compose.yml`, `nginx.conf`, android logcat, browser requests, etc.

<details><summary>Name of the information here</summary><br><pre>

contents here

</pre></details>

**Comment by jmattheis** [comment]
In your example how you created, you've defined `gotify` as database name. In the connection string it is `gotifydb`. Gotify tries to connect via 127.0.0.1, this should be the same as localhost, but maybe mysql wants you to use 127.0.0.1 instead. If this doesn't work for you, can you add your mysql config via gist to this issue?

**Comment by dutchguy69** [comment]
Thank you for your work on this project and your assistance for me with your reply.

It turned out that my bind address for mySQL server was not 127.0.0.1 but a local IP. Once that was changed I saw errors in SYSLOG indicating that mysql did not allow the user to create tables. So I updated my GRANT statement and after 3 attempts it was working.

The gotifydb was a typo on my part as the db is called gotify without the db part.

It seems as a minimum for mySQL/mariaDB you will have to execute the below commands before the service can be started. This is not needed for SQLite. This is not mentioned in the documentation and I think that is why I was uncertain about my connect string in the first place.

CREATE DATABASE gotifyDB;
CREATE USER 'gotify'@'localhost' IDENTIFIED BY 'SecretSecurePassword';
GRANT CREATE, SELECT, INSERT, UPDATE, DELETE, INDEX, ALTER ON gotifyDB.* TO 'gotify'@'localhost';

If you want to be less specific about access and security then the command can simply be

GRANT ALL ON gotifyDB.* TO 'gotify'@'localhost';

maybe someone can add this and or a warning about this in the documentation.
